packages/flutter/lib/src/material/mergeable_material.dart

```diff
@@ -122,10 +122,10 @@ class MergeableMaterial extends StatefulWidget {
   final bool hasDividers;
 
   @override
-  String toString() {
```

Comment:
in this case it does indeed need to be an int for the internals of the class to work correctly. Switched the diagnostic view of the property to show as a DoubleProperty as the fact that the value can only be an int shouldn't be visible in the debugging output.